File: cpp/Motorcycle.cpp
#include "Motorcycle.hpp"

// Constructor
Motorcycle::Motorcycle(const std::string &plate, const std::string &brand, int year,
                       const std::string &clr, int type, int tankCapacity)
    : Vehicle(plate, brand, clr, year), type(type), tankCapacity(tankCapacity) {
}

// Destructor
Motorcycle::~Motorcycle() {
}

// Getters
int Motorcycle::getType() const {
    return type;
}

int Motorcycle::getTankCapacity() const {
    return tankCapacity;
}

// Setters
void Motorcycle::setType(int type) {
    this->type = type;
}

void Motorcycle::setTankCapacity(int capacity) {
    tankCapacity = capacity;
}

File: python/Motorcycle.py
from Vehicle import Vehicle

class Motorcycle(Vehicle):
    def __init__(self, plate, brand, year, clr, type, tankCapacity):
        super().__init__(plate, brand, clr, year)
        self._type = type
        self._tankCapacity = tankCapacity

    # Getters
    def getType(self):
        return self._type

    def getTankCapacity(self):
        return self._tankCapacity

    # Setters
    def setType(self, type):
        self._type = type

    def setTankCapacity(self, capacity):
        self._tankCapacity = capacity
